libbeat/mapping/field.go

```diff
@@ -119,6 +122,55 @@ func (f *Field) Validate() error {
 	return nil
 }
 
+func (f *Field) validateType() error {
+	switch strings.ToLower(f.Type) {
+	case "text", "keyword":
+		return errors.Wrapf(stringType.validate(f.Format), "field %s", f.Name)
+	case "long", "integer", "short", "byte", "double", "float", "half_float", "scaled_float":
+		return errors.Wrapf(numberType.validate(f.Format), "field %s", f.Name)
+	case "date", "date_nanos":
+		return errors.Wrapf(dateType.validate(f.Format), "field %s", f.Name)
+	case "geo_point":
+		return errors.Wrapf(geoPointType.validate(f.Format), "field %s", f.Name)
+	case "boolean", "binary", "ip", "alias", "array":
+		if f.Format != "" {
+			return fmt.Errorf("no format expected for field %s, found: %s", f.Name, f.Format)
+		}
+	case "object", "group", "nested":
+		// No check for them yet
+	case "":
+		// Module keys, not used as fields
+	default:
+		// There are more types, not being used by beats, to be added if needed
+		return fmt.Errorf("unexpected type '%s' for field '%s'", f.Type, f.Name)
+	}
+	return nil
+}
+
+type fieldTypeGroup struct {
+	name    string
+	formats []string
+}
+
+var (
+	stringType   = fieldTypeGroup{"string", []string{"string", "url"}}
```

Comment:
How about ip? I'm not sure if I remembered correctly, I thought there is a type called IP 😬

Reply:
`ip`  is in the group of types with no formats expected a few lines up.

Reply:
👍 Thanks! I missed it!!